lib/storage/client.js

```diff
@@ -1182,6 +1183,267 @@ class Client extends CryptoConsumer {
+   *
+   * @param {string} groupId   UUID assigned by TozStore, used to identify a group.
+   * @param {Array} writerIds  Array of writerIds to filter responses by; optional.
+   * @param {number} nextToken  Indicates where to start pagination; optional.
+   * @param {number} max        The maximum number of groups to list per request; optional.
+   *
+   * @return {Array}  An array of records returned from TozStore.
+   */
+  async listRecordsSharedWithGroup(
+    groupId,
+    writerIds = [],
+    nextToken = null,
+    max = null
+  ) {
+    const listSharedRequest = {
+      group_id: groupId,
+      writer_ids: writerIds,
+      nextToken: nextToken,
+      max: max,
+    }
+    let encodedUrl = urlEncodeData(listSharedRequest)
+    let response = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups/${groupId}/share?${encodedUrl}`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    )
+    let listSharedWithGroups = []
+    let recordsJson = await validateResponseAsJSON(response)
+    if (recordsJson.results == null) {
+      return listSharedWithGroups
+    }
+    for (var i = 0; i < recordsJson.results.length; i++) {
+      let meta = await new Meta(
+        recordsJson.results[i].meta.writer_id,
+        recordsJson.results[i].meta.user_id,
+        recordsJson.results[i].meta.type,
+        recordsJson.results[i].meta.plain
+      )
+      let recordData = await new RecordData(recordsJson.results[i].record_data)
+      let record = await new Record(
+        meta,
+        recordData,
+        recordsJson.results[i].rec_sig
+      )
+      let decryptedRecord = shared.decryptRecord(this, record)
+      listSharedWithGroups.push(decryptedRecord)
+    }
+    return listSharedWithGroups
+  }
+
+  /**
+   * shareRecordWithGroup
+   *
+   * @param {string} groupId   UUID assigned by TozStore, used to identify a group.
+   * @param {string} recordType Record type to be shared with the group.
+   *
+   * @return {Promise<Record>}
+   */
+  async shareRecordWithGroup(groupId, recordType) {
+    // Get the Groups Public key
+    let groupInfo = this.readGroup(groupId)
+    // Get the encrypted Access Key
+    let encryptedAccessKey = await this.getReaderKey(
+      this.config.clientId,
+      this.config.clientId,
+      recordType
+    )
+    // Share with group
+    const shareRecordRequest = {
+      group_id: groupId,
+      record_type: recordType,
+      encrypted_access_key: encryptedAccessKey.eak,
+      public_key: groupInfo.publicKey,
+    }
+    let response = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups/${groupId}/share`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(shareRecordRequest),
+      }
+    )
+    let recordsJson = await validateResponseAsJSON(response)
```

Comment:
recordsJson doesn't reflect what's being returned here, also the assignment is unecessary